### ISSUE TITLE
Training Error when trying to reproduce results

### ISSUE THREAD

**Issue by derek-saal** [yao8839836/text_gcn#6]
I tried to run this in a google colab notebook:
> !git clone https://github.com/yao8839836/text_gcn.git

>import nltk
nltk.download('stopwords')

> !cd text_gcn; python ./remove_words.py

> !cd text_gcn; python ./build_graph.py

> !cd text_gcn; python ./train.py

This error was from the last step.

> Tensor("graphconvolution_2/SparseTensorDenseMatMul/SparseTensorDenseMatMul:0", shape=(?, 20), dtype=float32)
WARNING:tensorflow:From /content/text_gcn/metrics.py:6: softmax_cross_entropy_with_logits (from tensorflow.python.ops.nn_ops) is deprecated and will be removed in a future version.
Instructions for updating:
>
> Future major versions of TensorFlow will allow gradients to flow
into the labels input on backprop by default.
>
> See `tf.nn.softmax_cross_entropy_with_logits_v2`.
>
> 2018-10-14 05:20:29.837849: E tensorflow/stream_executor/cuda/cuda_driver.cc:300] failed call to cuInit: CUDA_ERROR_NO_DEVICE: no CUDA-capable device is detected
2018-10-14 05:20:29.837921: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:163] retrieving CUDA diagnostic information for host: 0d6b17cf0c02
2018-10-14 05:20:29.837949: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:170] hostname: 0d6b17cf0c02
2018-10-14 05:20:29.838019: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:194] libcuda reported version is: 396.44.0
2018-10-14 05:20:29.838109: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:198] kernel reported version is: 396.44.0
2018-10-14 05:20:29.838137: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:305] kernel version seems to match DSO: 396.44.0
2018-10-14 05:20:31.191157: W tensorflow/core/framework/allocator.cc:113] Allocation of 21592477600 exceeds 10% of system memory.
tcmalloc: large alloc 21592481792 bytes == 0xc6b28000 @  0x7f00ce49db6b 0x7f00ce4bd379 0x7f00a0e1bff7 0x7f00a0b1542f 0x7f00a0bb130a 0x7f00a0b85e21 0x7f00a0b8630b 0x7f00a0b863e8 0x7f00a32af18f 0x7f00a0d3b04d 0x7f00a0d80769 0x7f00a0d739e5 0x7f00a0dee692 0x7f00a0dec5d7 0x7f00ccd9d57f 0x7f00cde7f6db 0x7f00ce1b888f
2018-10-14 05:20:31.233049: W tensorflow/core/framework/op_kernel.cc:1273] OP_REQUIRES failed at sparse_tensor_dense_matmul_op.cc:145 : Invalid argument: k (4184079489906118640) from index[1014731,1] out of bounds (>=61603)
/bin/bash: line 1:   246 Segmentation fault      (core dumped) python ./train.py

**Comment by yao8839836** [comment]
@derek-saal 

Hi, thanks for trying to reproduce the results.

I think the main reason is that the memory is not enough for 20ng. I tried the code on my MacBook Pro (16G memory), and the same problem occurs. But when I run the same code on a server, It works fine.

The memory cost for R8, R52, ohsumed and mr is much lower, I think the code will works fine in the same environment when you use dataset = 'R8' or 'R52' or 'ohsumed' or 'mr' in remove_words.py, build_graph.py and train.py

**Comment by derek-saal** [comment]
Wow, fast response. Thank you for that @yao8839836 .

Ill try that and let you know what I find.

Thanks,
@derek-saal

**Comment by derek-saal** [comment]
Sorry for ^

I just meant to comment, not close and comment 😮

**Comment by derek-saal** [comment]
@yao8839836 , I ran MR and got these results:

> Test set results: cost= 0.49755 accuracy= 0.77068 time= 0.12100

Thanks again!

**Comment by yao8839836** [comment]
@derek-saal 

That is good, my pleasure!

**Comment by d04922013** [comment]
Hi, it seems that "GeForce GTX 1060 + 6G RAM" could not work for "R8." Is it possible to know your hardware spec as a reference? Thanks for publishing your code. It's really helpful.
